workflow/controller/scope_test.go

```diff
@@ -0,0 +1,184 @@
+package controller
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var s3Artifact = `
+name: s3-artifact
+path: some/local/path
+s3:
+  endpoint: minio:9000
+  bucket: test-bucket
+  key: /path/to/some/key
+  accessKeySecret:
+    key: accesskey
+    name: my-minio-cred
+  secretKeySecret:
+    key: secretkey
+    name: my-minio-cred
+`
+
+var s3ArtifactWithSubpath = `
+name: s3-artifact
+path: some/local/path
+s3:
+  endpoint: minio:9000
+  bucket: test-bucket
+  key: /path/to/some/key/some/subkey
+  accessKeySecret:
+    key: accesskey
+    name: my-minio-cred
+  secretKeySecret:
+    key: secretkey
+    name: my-minio-cred
+`
+
+var ArtifactoryArtifact = `
```

Comment:
these variable should all be private

Reply:
Addressed in 80dd46e9d67ad76397c5cd3ffecdb2159c4aff59